src/ToolBox/superpmi/superpmi-shim-collector/icorjitinfo.cpp

```diff
@@ -1118,6 +1118,13 @@ CorInfoType interceptor_ICJI::getFieldType(CORINFO_FIELD_HANDLE  field,
                                            )
 {
     mc->cr->AddCall("getFieldType");
+    // 'structType' may be null, but since we use only 'field' and 'memberParent' as keys,
+    // we don't want to pass null, in case we have a subsequent call with non-null 'structType'.
+    CORINFO_CLASS_HANDLE dummyStructType;
```

Comment:
I don't think this is the correct solution. Fundamentally, we should not alter the function arguments in the collector; we should always pass through what the JIT-passed arguments are intact, and return exactly what the VM returns.

I think in this case perhaps the best option is to change recGetFieldType in the non-null case to first check if the entry (null or not) exists in the LWM (using GetIndex?) and then replacing the stored value.

An alternative that might be better would be to make "structType==nullptr" (the bool condition) a key. Then, we'd store a null and non-null answer.

Reply:
Hmm, when I've implemented these I always have had the shim layer pass all the optional arguments. I suppose if the presence of arguments these alters the logic then that could perturb the results. And that could well be the case.

I wonder if we could have some sort of SPMI replay fidelity check where we diff SPMI replay versus the jit runs that collected the data?

I suppose it comes down to whether the args are altering API behavior or just